src/connector/src/parser/unified/debezium.rs

```diff
@@ -535,9 +540,554 @@ pub fn extract_bson_id(id_type: &DataType, bson_doc: &serde_json::Value) -> Acce
     Ok(id)
 }
 
+/// Extract the field data from the bson document
+///
+/// BSON document is a JSON object with some special fields, such as:
+/// long integer: {"$numberLong": "1630454400000"}
+/// date time: {"$date": {"$numberLong": "1630454400000"}}
+///
+/// For now, we support only the Canonical format of the date and timestamp.
+// similar to extract the "_id" field from the message payload
+pub fn extract_bson_field(
+    type_expected: &DataType,
+    bson_doc: &serde_json::Value,
+    field: Option<&str>,
```

Comment:
Add to doc comment above that `field` is `None` when accessing elements within an array.